### ISSUE TITLE
Version Number Change?

### ISSUE THREAD

**Issue by SomeCrazyGuy** [thesourcerer8/hddsuperclone#17]
Considering the number of changes in the last couple months, it may be time to change the version number.

**Comment by thesourcerer8** [comment]
Yes, I agree, I think we should produce a new version, and also provide binaries of the HDDScViewer for Windows. Is there anything else pending for the current release? I would suggest we build a release-candidate version, test it through, and if we pass all tests, we will provide a new official version.

**Comment by SomeCrazyGuy** [comment]
The only pending changes i want to merge for the current release is a makefile change to enable compiler optimizations and also fix the warnings that optimizations uncover. 

I think version 2.3.4 should provide a stable release that is compatible and familiar for users of the last proprietary version.

Maybe after that point create an hddsuperclone3 branch where i can merge some breaking / incompatible changes ive been prototyping.

**Comment by thesourcerer8** [comment]
Ok, let's get your compiler optimizations into the release, yes. I prefer a more stable release, and I think that we can gain some more stability from properly triaging warnings and cleaning up any open issues. Can I continue in the mean-time to work on the code, or should I leave it to you some more to ease merging your changes?

**Comment by thesourcerer8** [comment]
I would have thought to give it a more different version number like 2.4 at least, to more clearly differentiate it from the old closed-source versions, even if it's still compatible.

**Comment by SomeCrazyGuy** [comment]
Feel free to work on the code, I believe every warning caused by enabling optimizations is related to the string functions (like strncpy) probably because optimizations cause the compiler to replace those functions with builtins which allow the compiler to perform more checks on the code.

I think 2.4.0 is good

One last possible issue: gtk cannot be statically compiled, a windows build of the viewer will also have to ship an extra  18mb of gtk dlls as well.

**Comment by thesourcerer8** [comment]
Hmm, that isn't perfect, but if there is no way around it, I don't mind packaging 18MB of GTK DLLs with it. Can you provide a package with them? I would like to take a look at it then.

**Comment by SomeCrazyGuy** [comment]
I will send a link after work when I can get to my computer. I might look into the possibility of building something using a native ui on windows or something with a smaller cross-platform footprint like SDL.

**Comment by SomeCrazyGuy** [comment]
Here is my build of hddscviewer: https://drive.google.com/file/d/12UwjPnnvFhud4XhLLJ2Mx955ubWtb0oa/view?usp=sharing
